examples/ransac_compare.py

```diff
@@ -0,0 +1,213 @@
+"""
+=================
+RANSAC COMPARE
+=================
+
+
+In this example we compare the RANSAC implementations of pyprep and autoreject.
+
+.. currentmodule:: pyprep
+"""  # noqa: D205 D400
+
+# Authors: Yorguin Mantilla <yjmantilla@gmail.com>
+#
+# License: MIT
+# Based On: run_ransac.py and run_full_prep.py
+
+###############################################################################
+# First we import what we need for this example.
+import numpy as np
+import mne
+from scipy import signal as signal
+from time import perf_counter
+from autoreject import Ransac  # noqa
+from autoreject.utils import interpolate_bads  # noqa
+from mne.utils import check_random_state
+import pyprep.ransac as ransac_pyprep
+
+
+###############################################################################
+# Now let's make some arbitrary MNE raw object for demonstration purposes.
```

Comment:
why not use this function for generating a random mne object? https://github.com/sappelhoff/pyprep/blob/8476ea5c1f80c60e564ce8e4c2fca34464515b40/pyprep/utils.py#L93

Reply:
Im not yet sure it will work without tweaking the function inside a bit. Since the frequency components are selected randomly from a range inside the function I cannot guarantee easily that the good channels will correlate highly each with each other.

We could make a function to get an arbitrary harmonic signal by summing arbitrary frequencies, amplitudes and phases , though , is it worth it? Something of this sort:

```python
def get_harmonic_signal(freqs,weights,phases,s_rate=1000,start=0,stop=10,withtime=False,time=None):
    """
    Function that returns a harmonic signal given its components.
    Parameters:
        freqs: list of floats
                the frequencies [Hz] of the components
        weights: list of floats
                the weights  of the components
        phases: list of floats
                the phases of the components
        s_rate: float, default=1000
                the sampling rate
        start: float, default=0
                start time
        stop: float, default=10
                stop time
        withtime: bool, default=False
                Whether to return or not
                the time vector of the signal.
        time : np.ndarray, default=None
                1d time vector. If None
                it is created from the start,
                stop and s_rate parameters.
    Returns:
        if withtime:
            time,harmonic_signal
        else:
            harmonic_signal
        harmonic_signal: numpy.ndarray
                the harmonic signal generated
        
    Example:
    >>>s_rate = 1000
    >>>length = s_rate
    >>>freqs_sin = np.array([10,10])
    >>>weights_sin = np.array([1,1])
    >>>phases_sin = np.array([0,90])
    >>>start = 0
    >>>stop = 10
    >>>sin = get_harmonic_signal(freqs_sin,weights_sin,phases_sin,s_rate,start,stop)
    """
    freqs = as_list(freqs)
    weights = as_list(weights)
    phases = as_list(phases)
    if time is None:
        time = np.arange(start, stop, 1/s_rate)
    time = np.squeeze(time)
    harmonic_signal = np.zeros(time.shape[0])
    rad_phases = [phase * np.pi/180 for phase in phases]
    rad_freqs = [2 * np.pi * freq for freq in freqs]
    sinusoid = np.zeros((len(freqs),time.shape[0]))
    for i in range(0,len(freqs)):
        sinusoid[i,:] = weights[i] * np.sin(rad_freqs[i] * time + rad_phases[i])

    harmonic_signal = np.sum(sinusoid,axis=0)
    if withtime == True:
        return time,harmonic_signal
    else:
        return harmonic_signal

```

Reply:
mmh I take back what I said: It's probably not worth it to make a new big function for that.